src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -245,25 +257,32 @@ protected int doWork() {
                 }
             }
         } catch (IOException e) {
-            throw new PicardException("Can't write to or close output file " + OUTPUT.getAbsolutePath(), e);
+            throw new PicardException("Can't write to or close output file " + OUTPUT.getURIString(), e);
         } catch (IllegalArgumentException e) {
             // in case of an unexpected error delete the file so that there isn't a
             // truncated result which might be valid yet wrong.
-            OUTPUT.delete();
+            OUTPUT.toPath().toFile().delete();
+            if (Files.exists(OUTPUT.toPath())){
+                try {
+                    Files.delete(OUTPUT.toPath());
+                } catch (IOException e2) {
+                    throw new PicardException("Unknown problem encountered, and failed to delete the incomplete output.", e);
+                }
+            }
```

Comment:
I don't recall if we talked about this previously, but why is this second deletion attempt necessary ?

Reply:
I don't think it's necessary; I removed the first attempt, `OUTPUT.toPath().toFile().delete();`, which as you pointed out probably doesn't work on a remote path.